### PR TITLE
Add Entry: C++ Maps 

### PR DESCRIPTION
**Pull request by KTom101**
New entry on C++ Maps and terms; keyword 19.

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/cpp/concepts/maps/maps.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+To set the comparison function while intializing an empty map:
+
+```pseudo
+std::map <type1, type2, std::greater<dataType>> mapName;
+```
+
+To set the comparison function while intializing a map with items:
+
+```pseudo
+map<type1, type2,  std::greater<dataType>> mapName { {key1, value1}, {key2, value2}, ...};
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:121]:
```suggestion
std::map <type1, type2,  std::greater<dataType>> mapName { {key1, value1}, {key2, value2}, ...};
```

File: content/cpp/concepts/maps/maps.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+---
+
+Maps are associative containers that have elements with key, value pairs. The keys are used to sort and identify the elements, while the values store the content associated with the keys. Each mapped value must have a unique key value.
+
+## Syntax
+
+An empty map can be created by using the map keyword, declaring the data types of the key and value, and setting a `mapName`:
+
+```pseudo
+map<type1, type2> mapName;
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:23]:
```suggestion
std::map <type1, type2> mapName;
```

File: content/cpp/concepts/maps/maps.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+`type1` and `type2` are the data types of the key and value, respectively.
+
+To set a list at declaration, the following syntax is used:
+
+```pseudo
+map<type1, type2> mapName { {key1, value1}, {key2, value2}, ...};
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:31]:
```suggestion
std::map <type1, type2> mapName { {key1, value1}, {key2, value2}, ...};
```

File: content/cpp/concepts/maps/maps.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+The following example creates an empty map, `emptyMap`, and a map set with values, `clothingStore`:
+
+```cpp
+#include <iostream>
+#include <iterator>
+#include <map>
+
+int main(){
+  // Initializing empty map
+  std::map<std::string, int> emptyMap;
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:47]:
```suggestion
  std::map <std::string, int> emptyMap;
```

File: content/cpp/concepts/maps/maps.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+  // Initializing map with items
+  std::map<std::string, int> clothingStore {{"tshirt", 10}, {"pants", 12}, {"sweaters", 18}};
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:50]:
```suggestion
  std::map <std::string, int> clothingStore {{"tshirt", 10}, {"pants", 12}, {"sweaters", 18}};
```

File: content/cpp/concepts/maps/maps.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+This returns the mapped value associated with the `key`.
+
+### Example
+
+The `sweaters` element is retrieved from the `clothingStore` map initialized in the previous example:
+
+```cpp
+std::cout<< clothingStore["sweaters"]; // Output: 18
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:69]:
```suggestion
std::cout << clothingStore["sweaters"]; // Output: 18
```

File: content/cpp/concepts/maps/maps.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+
+int main(){
+
+ // Initializing map with items
+ std::map <int, std::string> reptiles {
+   {10, "Komodo Dragon"}, {15, "Saltwater Crocodile"}, {8, "Leatherback Sea Turtle"} };
+
+
+  // Initializing iterator
+  std::map<int, std::string> :: iterator iter;
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:89]:
```suggestion
  std::map <int, std::string> :: iterator iter;
```

File: content/cpp/concepts/maps/terms/clear/clear.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+```codebyte/cpp
+#include <iostream>
+#include <iterator>
+#include <map>
+
+int main(){
+
+  // Initializing map with items
+  std::map<std::string, int> supplies {{"pen", 5}, {"pencils", 3}, {"paper", 12}};
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/clear/clear.md:36]:
```suggestion
  std::map <std::string, int> supplies {{"pen", 5}, {"pencils", 3}, {"paper", 12}};
```

File: content/cpp/concepts/maps/terms/clear/clear.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+
+  // Clear items
+  supplies.clear();
+
+  // Print size of map
+  std::cout<<supplies.size();
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/clear/clear.md:42]:
```suggestion
  std::cout << supplies.size();
```

File: content/cpp/concepts/maps/terms/insert/insert.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  std::map<std::string, int> lifeSpan;
+
+  // Adding elements
+  lifeSpan.insert(std::pair<std::string, int>("Giraffe", 26));
+  lifeSpan.insert(std::pair<std::string, int>("Goat", 15));
+  lifeSpan.insert(std::pair<std::string, int>("Lion", 10));
+  lifeSpan.insert(std::pair<std::string, int>("Tiger", 8));
+
+  // Initializing iterator
+  std::map<std::string, int> :: iterator iter;
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/insert/insert.md:44]:
```suggestion
  std::map <std::string, int> :: iterator iter;
```

File: content/cpp/concepts/maps/terms/remove/remove.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+Tags:
+  - 'Objects'
+  - 'OOP'
+  - 'Classes'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
+
+The `.erase()` method removes an element by it's key from a map.
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:16]: Is it `.erase()` or `.remove()`?

File: content/cpp/concepts/maps/terms/remove/remove.md [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+The following codebyte example uses `.erase()` to remove an element from the `fields` map:
+
+```cpp
+#include <iostream>
+#include <iterator>
+#include <map>
+
+int main(){
+  // Initializing map with items
+  std::map<std::string, std::string> fields {{"zero", "0"}, {"one", "1"},  {"two", "2"}, {"three", "3"}};
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:37]:
```suggestion
  std::map <std::string, std::string> fields {{"zero", "0"}, {"one", "1"},  {"two", "2"}, {"three", "3"}};
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@KTom101, reviewed and left a few comments.

File: content/cpp/concepts/maps/maps.md [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+```cpp
+#include <iostream>
+#include <iterator>
+#include <map>
+
+int main(){
+
+ // Initializing map with items
+ std::map <int, std::string> reptiles {
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:84]:
```suggestion
 std::map<int, std::string> reptiles {
```

File: content/cpp/concepts/maps/maps.md [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+```
+
+The default comparison function can be changed to `std::greater<dataType>` in order to sort the elements in descending order.
+
+### Syntax
+
+To set the comparison function while intializing an empty map:
+
+```pseudo
+std::map <type1, type2, std::greater<dataType>> mapName;
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:115]:
```suggestion
std::map<type1, type2, std::greater<dataType>> mapName;
```

File: content/cpp/concepts/maps/maps.md [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+
+```codebyte/cpp
+#include <iostream>
+#include <iterator>
+#include <map>
+
+int main(){
+
+ // Initializing map with items
+ std::map <int, std::string, std::greater<int>> reptiles {
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:138]:
```suggestion
 std::map<int, std::string, std::greater<int>> reptiles {
```

File: content/cpp/concepts/maps/terms/remove/remove.md [hunk 16]
```diff
@@ -0,0 +1,52 @@
+---
+Title: '.remove()'
```
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:2]:
```suggestion
Title: '.erase()'
```

Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:2]: If this is `.erase()` the file needs to be renamed.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]

**Comment by KTom101** [comment]
@SSwiniarski I believe I made all your changes including the one about the file name. Please make sure, though. I had an issue where some of the direct commits didn't go through. So I made some of the changes locally. Thanks!

**Review by SSwiniarski — APPROVED** [review]
Looks good for next review.

File: content/cpp/concepts/maps/terms/insert/insert.md [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+---
+
+Elements can be added to a map by inserting key-value pairs using the `.insert()` method.
+
+## Syntax
+
+```pseudo
+mapName.insert(std::pair<type1, type2>(key, value));
+```
+
```
Comment (Dusch4593) [inline content/cpp/concepts/maps/terms/insert/insert.md:23]: Could we add a sentence or two here about the snippet above?

File: content/cpp/concepts/maps/terms/clear/clear.md [hunk 18]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+The `.clear()` method removes all elements of a map. This destroys the elements in the container making the new size of the container to be 0.
+
+## Syntax
+
+```pseudo
+mapName.clear()
+```
+
```
Comment (Dusch4593) [inline content/cpp/concepts/maps/terms/clear/clear.md:23]: Could we add a sentence or two here about the snippet above?

File: content/cpp/concepts/maps/maps.md [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+
+To set the comparison function while intializing a map with items:
+
+```pseudo
+std::map<type1, type2, std::greater<dataType>> mapName { {key1, value1}, {key2, value2}, ...};
+```
+
+- `dataType` for the comparison function must be the same as `type1` which is the data type for the keys.
+
+### Example
```
Comment (Dusch4593) [inline content/cpp/concepts/maps/maps.md:126]:
```suggestion
### Codebyte Example
```

File: content/cpp/concepts/maps/maps.md [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+std::map<type1, type2, std::greater<dataType>> mapName;
+```
+
+To set the comparison function while intializing a map with items:
+
+```pseudo
+std::map<type1, type2, std::greater<dataType>> mapName { {key1, value1}, {key2, value2}, ...};
+```
+
+- `dataType` for the comparison function must be the same as `type1` which is the data type for the keys.
```
Comment (Dusch4593) [inline content/cpp/concepts/maps/maps.md:124]:
```suggestion
The `dataType` for the comparison function above must be the same as `type1` which is the data type for the keys.
```

File: content/cpp/concepts/maps/maps.md [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+  for (iter = reptiles.begin(); iter != reptiles.end(); ++iter)
+     {
+      std::cout << '\t' << iter->first << '\t' << iter->second
+      << '\n';
+    }
+
+}
+```
+
+The above outputs the following:
```
Comment (Dusch4593) [inline content/cpp/concepts/maps/maps.md:100]:
```suggestion
The snippet above outputs the following:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@KTom101 Finished with my review and left some comments/suggestions.

**Comment by KTom101** [comment]
@Dusch4593 I think I've finished with your changes. Let me know if you want to switch things around.

**Review by Dusch4593 — APPROVED** [review]
@KTom101 LGTM! 👍🏻

**Comment by Dusch4593** [comment]
@yangc95 @caupolicandiaz does this PR need one more review before merge?

**Review by caupolicandiaz — COMMENTED** [review]
LGTM!

**Comment by Dusch4593** [comment]
Thank you so much for these new entries @KTom101! 🚀

Once the entries go live we can check them out with the links below:
C++: Maps - https://www.codecademy.com/resources/docs/cpp/maps
C++ Maps: .clear() - https://www.codecademy.com/resources/docs/cpp/maps/clear
C++ Maps: .erase() - https://www.codecademy.com/resources/docs/cpp/maps/erase
C++ Maps: .insert() - https://www.codecademy.com/resources/docs/cpp/maps/insert

**Pull request merged**